docs/documentation/namespaces-proposal.md

```diff
@@ -0,0 +1,93 @@
+# Namespaces Proposal
+
+This document contains a proposal for a guideline on how to expand the usage of 
+php namespaces to ILIAS modules and services. The suggested approach will elaborate 
+the following aspects:
+* naming conventions
+* integration of composer and autoloading
+* handling of subdirectories and interfaces
+* migration of the current structure
+* possible problems
+
+## Guideline
+
+### Naming & Structuring
+
+* The naming and structuring of all Namespaces MUST follow the PHP Standards Recommendation PSR-4: https://www.php-fig.org/psr/psr-4/.
+* The base prefix namespace is *ILIAS* and is pointing at the subdirectory */src*. Therefore, all Modules and Services MUST be located in the directory */src*. 
+* The subdirectory */classes* for Modules and Services will not be needed anymore and MUST be removed.
+* The prefixes "*class.*" and "*il*" for ILIAS classes will not be needed anymore and MUST be removed.
+* Newly created interfaces, abstract classes and traits MUST follow the PSR Naming Conventions: https://www.php-fig.org/bylaws/psr-naming-conventions. Already existing interfaces, abstract classes and traits SHOULD be renamed.
+* Every Module and Service folder MUST have a subdirectory for every type of asset (js, css, templates, images, docs) 
```

Comment:
Will there be any convention that will document this for the future? For example `js` stands for `JavaScript`. Without any clicking to the project source you won't be able to know what the subfolder is named. IMO there must be a document which lists all the common file types and the associated file names. New file types must be proposed/discussed to Jour Fixe(?) or the Technical Board(?), and a appropriate subfolder is added for this file/asset type .